### ISSUE TITLE
Unable to install 1.8.0 on bare metal k8s 1.18.5

### ISSUE THREAD

**Issue by briantopping** [cilium/cilium#12385]
## Bug report

**General Information**

- Cilium version (run `cilium version`): 1.18.0
- Kernel version (run `uname -a`): 5.6.14-1.el7.elrepo.x86_64
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
```
[root@r3c1n1 kubeadm]# kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:47:41Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"clean", BuildDate:"2020-06-26T03:39:24Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Link to relevant artifacts (policies, deployments scripts, ...): Installed via `kubeadm init --config config.yaml `
```
apiVersion: kubeadm.k8s.io/v1beta2
kind: InitConfiguration
bootstrapTokens:
- groups:
  - system:bootstrappers:kubeadm:default-node-token
  token: to.ken
[cilium-sysdump-20200702-092541.zip](https://github.com/cilium/cilium/files/4865257/cilium-sysdump-20200702-092541.zip)


  ttl: 24h0m0s
  usages:
  - signing
  - authentication
localAPIEndpoint:
  advertiseAddress: 10.10.8.31
  bindPort: 6443
nodeRegistration:
  criSocket: /var/run/dockershim.sock
  name: r3c1n1.example.com
---
apiVersion: kubeadm.k8s.io/v1beta2
kind: ClusterConfiguration
apiServer:
  timeoutForControlPlane: 4m0s
  certSANs:
  - "r3c1.example.com"
  - "r3c1n1.example.com"
  - "r3c1n2.example.com"
  - "r3c1n3.example.com"
  - "10.10.8.32"
  - "10.10.8.33"
  - "127.0.0.1"
certificatesDir: /etc/kubernetes/pki
clusterName: "r3c1"
controllerManager: {}
controlPlaneEndpoint: "10.10.8.30:6444"
dns:
  type: CoreDNS
etcd:
  local:
    dataDir: /var/lib/etcd
imageRepository: k8s.gcr.io
kubernetesVersion: v1.18.5
networking:
  dnsDomain: r3c1.example.com
  podSubnet: 10.16.0.0/14
  serviceSubnet: 10.96.0.0/14
scheduler: {}
```
- Generate and upload a system zip:
```
curl -sLO https://git.io/cilium-sysdump-latest.zip && python cilium-sysdump-latest.zip
```
This didn't quite work, gave an error. See attached with what was generated:
```
INFO collected cilium configmap yaml file: cilium-configmap-20200702-092546.yaml
INFO collecting cilium-bugtool output ...
INFO collected cilium-bugtool output: bugtool-cilium-9rvxf-20200702-092547.tar
INFO collecting cilium logs ...
INFO collected log file: cilium-9rvxf-20200702-092549
DEBUG no previous pod logs to gather for pod/container cilium-9rvxf/cilium-agent
INFO collected log file: cilium-operator-754dd76d85-vb9j5-20200702-092550
Traceback (most recent call last):
  File "/usr/lib64/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "cilium-sysdump-latest.zip/__main__.py", line 150, in <module>
  File "cilium-sysdump-latest.zip/sysdumpcollector.py", line 534, in collect
  File "cilium-sysdump-latest.zip/sysdumpcollector.py", line 109, in collect_logs
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 250, in map
    return self.map_async(func, iterable, chunksize).get()
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 554, in get
    raise self._value
KeyError: 'containerStatuses'
```

**How to reproduce the issue**

1. Install k8s 1.18.5 on bare metal via the above configuration and `kubeadm init --config config.yaml`
1. Install Cilium via `kubectl create -f https://raw.githubusercontent.com/cilium/cilium/1.8.0/install/kubernetes/quick-install.yaml`
1. cilium will load, but never gets healthy
1. Logs report following:
```
level=info msg="Retrieved node information from cilium node" nodeName=r3c1n1.hndc1.coglative.com subsys=k8s
level=warning msg="Waiting for k8s node information" error="required IPv4 pod CIDR not present in node resource" subsys=k8s
```

**Additional Attempts**

After a couple of days of failures and generally trying to debug what was going on, I tried various versions:
* k8s v1.18.5 + Cilium 1.8 : as reported
* k8s v1.17.8 + Cilium 1.8 : as reported
* k8s v1.17.8 + Cilium 1.7 : works correctly

Thanks for your amazing work!

**Comment by aanm** [comment]
@briantopping what's the output of `kubect get pods -n kube-system -o wide` and `kubectl logs --timestmaps -n kube-system cilium-operator-xxxx`?

**Comment by briantopping** [comment]
@aanm That provided the cue I needed to solve this: There seems to be a new dependency on `cilium-operator` from `cilium`. I wasn't used to removing the master taint at this point in my bootstrap, so `cilium-operator` was not loading.

Once the master taint was removed, things continued loading as normal.

Not sure if this is an unavoidable dependency, but if so, documenting it may be helpful.

Thanks!
